test/metabase/db/custom_migrations_test.clj

```diff
@@ -1033,3 +1039,444 @@
+                [dashcard-id]  (t2/insert-returning-pks! :model/DashboardCard {:dashboard_id           dashboard-id
+                                                                               :visualization_settings dashcard-vis
+                                                                               :card_id                card-id
+                                                                               :size_x                 4
+                                                                               :size_y                 4
+                                                                               :col                    1
+                                                                               :row                    1})]
+            (let [expected-settings {:graph.dimensions ["CREATED_AT" "CATEGORY"],
+                                     :graph.metrics    ["count"],
+                                     :click            "link",
+                                     :click_link_template
+                                     "http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}"
+                                     :click_behavior
+                                     {:type         "link",
+                                      :linkType     "url",
+                                      :linkTemplate "http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}"},
+                                     :column_settings
+                                     ;; the model keywordizes the json parsing yielding this monstrosity below
+                                     {"[\"ref\",[\"field\",2,null]]"
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com/{{ID}}",
+                                        :linkTextTemplate "here's an id: {{ID}}"}},
+                                      "[\"ref\",[\"field\",6,null]]"
+                                      {:click_behavior
+                                       {:type             "link",
+                                        :linkType         "url",
+                                        :linkTemplate     "http://example.com//{{id}}",
+                                        :linkTextTemplate "here is my id: {{id}}"}}}}]
+              (f)
+              (is (= expected-settings
+                     (-> (t2/select-one :model/DashboardCard :id dashcard-id)
+                         :visualization_settings))))))]
+    (testing "Running the migration from scratch"
+      (impl/test-migrations ["v48.00-022"] [migrate!]
+        (expect-correct-settings! migrate!)))
+    (testing "Running the migration after a previous data-migration still works"
```

Comment:
is this test meant to check if `migrate-click-through!` is idempotent?

either way I still think we need a test to check if `select count(*) from data_migrations where id = 'migrate-click-through'` returns 1, the custom migration `custom-migrations/MigrateClickThrough` will not be executed.